lib/src/rules/implicit_call_tearoff.dart

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2022, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+
+import '../analyzer.dart';
+
+const _desc =
+    r'Explicitly tear-off `call` methods when using an object as a Function.';
+
+const _details = r'''
+**DO** ...
+Explicitly tear off `.call` methods from objects when assigning to a Function
+type.
+
```

Comment:
Consider a sentence of rationale here?  (Maybe cribbed from the PR description which I found super edifying!)

Reply:
Done.